docs/development/priority-classes.md

```diff
@@ -0,0 +1,48 @@
+# `PriorityClasses` in Gardener Clusters
+
+Gardener makes use of [`PriorityClasses`](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/) to improve overall robustness of the system.
+In order to benefit from the full potential of `PriorityClasses`, gardenlet manages a set of well-known `PriorityClasses` with fine-granular priority values.
+
+All components of the system should use these well-known `PriorityClasses` instead of creating and using separate ones with arbitrary values, which would compromise the overall goal of using `PriorityClasses` in the first place.
+Gardenlet manages the well-known `PriorityClasses` listed in this document, so that third parties (e.g., gardener extensions) can rely on them to be present when deploying components to Seed and Shoot clusters.
```

Comment:
```suggestion
Gardenlet manages the well-known `PriorityClasses` listed in this document, so that third parties (e.g., Gardener extensions) can rely on them to be present when deploying components to Seed and Shoot clusters.
```